File: app/src/main/java/de/bornholdtlee/defaultproject/utils/SharedPreferencesUtils.java
package de.bornholdtlee.defaultproject.utils;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.HashSet;
import java.util.Set;

import de.bornholdtlee.defaultproject.R;

public class SharedPreferencesUtils {

    private final SharedPreferences sharedPreferencesReader;
    private final SharedPreferences.Editor sharedPreferencesWriter;

    public SharedPreferencesUtils(Context context) {
        sharedPreferencesReader = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE);
        sharedPreferencesWriter = sharedPreferencesReader.edit();
    }

    public void putString(String key, String value) {
        sharedPreferencesWriter.putString(key, value);
        sharedPreferencesWriter.apply();
    }

    public void putBoolean(String key, boolean value) {
        sharedPreferencesWriter.putBoolean(key, value);
        sharedPreferencesWriter.apply();
    }

    public void putInt(String key, int value) {
        sharedPreferencesWriter.putInt(key, value);
        sharedPreferencesWriter.apply();
    }

    public void putLong(String key, Long value) {
        sharedPreferencesWriter.putLong(key, value);
        sharedPreferencesWriter.apply();
    }

    public void remove(String key) {
        sharedPreferencesWriter.remove(key);
        sharedPreferencesWriter.apply();
    }

    public void removeAll() {
        sharedPreferencesWriter.clear().apply();
    }

    public String getString(String key, String defaultValue) {
        return sharedPreferencesReader.getString(key, defaultValue);
    }

    public boolean getBoolean(String key, boolean defaultValue) {
        return sharedPreferencesReader.getBoolean(key, defaultValue);
    }

    public long getLong(String key, long defaultValue) {
        return sharedPreferencesReader.getLong(key, defaultValue);
    }

    public int getInt(String key, int defaultValue) {
        return sharedPreferencesReader.getInt(key, defaultValue);
    }

    public void putFloat(String key, float value) {
        sharedPreferencesWriter.putFloat(key, value);
        sharedPreferencesWriter.apply();
    }

    public float getFloat(String key, float defaultValue) {
        return sharedPreferencesReader.getFloat(key, defaultValue);
    }

    public void addToKeySet(String key, String setEntry) {
        Set<String> set = getKeySet(key);
        set.add(setEntry);
        sharedPreferencesWriter.putStringSet(key, set);
        sharedPreferencesWriter.apply();
    }

    public HashSet<String> getKeySet(String key) {
        return new HashSet<>(sharedPreferencesReader.getStringSet(key, new HashSet<String>()));
    }
}

File: app/src/main/java/de/bornholdtlee/defaultproject/utils/SharedPreferencesUtils.kt
package de.bornholdtlee.defaultproject.utils

import android.content.Context
import android.content.SharedPreferences

import java.util.HashSet

import de.bornholdtlee.defaultproject.R

class SharedPreferencesUtils(context: Context) {

    private val sharedPreferencesReader: SharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
    private val sharedPreferencesWriter: SharedPreferences.Editor

    init {
        sharedPreferencesWriter = sharedPreferencesReader.edit()
    }

    fun putString(key: String, value: String) {
        sharedPreferencesWriter.putString(key, value)
        sharedPreferencesWriter.apply()
    }

    fun putBoolean(key: String, value: Boolean) {
        sharedPreferencesWriter.putBoolean(key, value)
        sharedPreferencesWriter.apply()
    }

    fun putInt(key: String, value: Int) {
        sharedPreferencesWriter.putInt(key, value)
        sharedPreferencesWriter.apply()
    }

    fun putLong(key: String, value: Long?) {
        sharedPreferencesWriter.putLong(key, value!!)
        sharedPreferencesWriter.apply()
    }

    fun remove(key: String) {
        sharedPreferencesWriter.remove(key)
        sharedPreferencesWriter.apply()
    }

    fun removeAll() {
        sharedPreferencesWriter.clear().apply()
    }

    fun getString(key: String, defaultValue: String): String? {
        return sharedPreferencesReader.getString(key, defaultValue)
    }

    fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return sharedPreferencesReader.getBoolean(key, defaultValue)
    }

    fun getLong(key: String, defaultValue: Long): Long {
        return sharedPreferencesReader.getLong(key, defaultValue)
    }

    fun getInt(key: String, defaultValue: Int): Int {
        return sharedPreferencesReader.getInt(key, defaultValue)
    }

    fun putFloat(key: String, value: Float) {
        sharedPreferencesWriter.putFloat(key, value)
        sharedPreferencesWriter.apply()
    }

    fun getFloat(key: String, defaultValue: Float): Float {
        return sharedPreferencesReader.getFloat(key, defaultValue)
    }

    fun addToKeySet(key: String, setEntry: String) {
        val set = getKeySet(key)
        set.add(setEntry)
        sharedPreferencesWriter.putStringSet(key, set)
        sharedPreferencesWriter.apply()
    }

    fun getKeySet(key: String): HashSet<String> {
        return HashSet(sharedPreferencesReader.getStringSet(key, HashSet())!!)
    }
}